cabal-helper.cabal

```diff
@@ -56,16 +56,22 @@ library
                        Paths_cabal_helper
   ghc-options:         -Wall
   build-depends:       base          < 5    && >= 4.5
-                     , Cabal         < 2.1  && >= 2.0     || < 1.26 && >= 1.14
+  if os(windows)
+    build-depends:     base                 && >= 4.7
+  build-depends:       Cabal         < 2.1  && >= 2.0     || < 1.26 && >= 1.14
                      , directory     < 1.4  && >= 1.1.0.2
                      , filepath      < 1.5  && >= 1.3.0.0
                      , transformers  < 0.6  && >= 0.3.0.0
                      , mtl           < 2.3  && >= 2.0
                      , process       < 1.7  && >= 1.1.0.1
-                     , unix          < 2.8  && >= 2.5.1.1
-                     , semigroupoids < 5.3  && >= 5.2.1
+  if os(windows)
+    build-depends:     unix-compat   < 0.5  && >= 0.4.3.1
```

Comment:
Since unix-compat re-exports stuff from unix on non-DOS patforms we might as well only depend on unix-compat and get rid of this conditional entirely. That should also get rid of the need for CPP in the imports.

Reply:
Unix-compat doesn't export Environment at all, so we can't get rid of unix. I wonder why not use System.Environment.setEnv from base? Is it because it requires base >=4.7?

Reply:
Yeah, I'd like to support everything back to GHC 7.4, which comes with base-4.5, since there hasn't been a good reason not to so far and doing that covers the most Cabal versions. Having tighter bounds on windos is ok though.

In that case just leave it in.

Reply:
And i just realized it's probably neccesary to bring unix-compat even on non-win platform, to get rid of ifdefs:
- Either win platform uses 4.7 base for setEnv and unix-compat for everything else and there is ifdefs in Compiletime.Data to use unix-compat on win and unix everywhere else + ifs in .cabal to exclude unix-compat fron non-win dependencies.
- Or we bring unix and unix-compat on both platforms, use unix-compat in Compiletime.data so no ifdefs there and ifs in .cabal. Ifdefs are only left in Compiletime.Compat.

Reply:
Using unix-compat on both platforms is fine with me. If it gets rid of conditionals in the cabal file and the source even better.